src/backend/crud/agent.py

```diff
@@ -74,16 +84,26 @@ def get_agents(
     """
     query = db.query(Agent)
 
+    if visibility == AgentVisibility.PUBLIC:
+        # Public agents -> is_private = False or None
+        query = query.filter((Agent.is_private == False) | (Agent.is_private == None))
```

Comment:
can remove None check 

re: what TJ said about removing nullable as we set the default to false.

nit: can remove these comments, the code is pretty self explanatory.